Group WKWebView state which is per-process into a struct

smfr opened the pull request:
#### ba12984f136d1a6f3560059f3df996fac9683e14
<pre>
Group WKWebView state which is per-process into a struct
<a href="https://bugs.webkit.org/show_bug.cgi?id=243704">https://bugs.webkit.org/show_bug.cgi?id=243704</a>

Reviewed by Chris Dumez.

We often have bugs where we forget to reset state on WKWebView when the web process terminates (the
same is true on WebPageProxy, which I will fix separately).

To make it harder to create such bugs, group all the data that needs to be reset on web process
termination into a PerWebProcessState struct. `_processWillSwapOrDidExit` can then just reinitialize
this struct.

Currently PerWebProcessState is only used for iOS, but it&apos;s probable that future refactoring will
make it apply to all platforms.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
(-[WKWebView _internalDoAfterNextPresentationUpdate:withoutWaitingForPainting:withoutWaitingForAnimatedResize:]):
(-[WKWebView _setMinimumEffectiveDeviceWidth:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _setHasCustomContentView:loadedMIMEType:]):
(-[WKWebView _willInvokeUIScrollViewDelegateCallback]):
(-[WKWebView _didInvokeUIScrollViewDelegateCallback]):
(-[WKWebView _resetCachedScrollViewBackgroundColor]):
(-[WKWebView _updateScrollViewBackground]):
(-[WKWebView _processWillSwapOrDidExit]):
(-[WKWebView _didRelaunchProcess]):
(-[WKWebView _didCommitLoadForMainFrame]):
(-[WKWebView _didCommitLayerTreeDuringAnimatedResize:]):
(-[WKWebView _trackTransactionCommit:]):
(-[WKWebView _restoreScrollAndZoomStateForTransaction:]):
(-[WKWebView _didCommitLayerTree:]):
(-[WKWebView _layerTreeCommitComplete]):
(-[WKWebView _restorePageScrollPosition:scrollOrigin:previousObscuredInset:scale:]):
(-[WKWebView _restorePageStateToUnobscuredCenter:scale:]):
(-[WKWebView _scrollToContentScrollPosition:scrollOrigin:animated:]):
(-[WKWebView _zoomToInitialScaleWithOrigin:animated:]):
(-[WKWebView _initialScaleFactor]):
(-[WKWebView _allowsDoubleTapGestures]):
(-[WKWebView _visibleContentRect]):
(-[WKWebView _dispatchSetViewLayoutSize:]):
(-[WKWebView _dispatchSetDeviceOrientation:]):
(-[WKWebView _beginAutomaticLiveResizeIfNeeded]):
(-[WKWebView _updateLiveResizeTransform]):
(-[WKWebView _frameOrBoundsChanged]):
(-[WKWebView _addUpdateVisibleContentRectPreCommitHandler]):
(-[WKWebView _scheduleVisibleContentRectUpdateAfterScrollInView:]):
(-[WKWebView _shouldDeferGeometryUpdates]):
(-[WKWebView _updateVisibleContentRects]):
(-[WKWebView _cancelAnimatedResize]):
(-[WKWebView _didCompleteAnimatedResize]):
(-[WKWebView _didStopDeferringGeometryUpdates]):
(-[WKWebView _keyboardChangedWithInfo:adjustScrollView:]):
(-[WKWebView _navigationGestureDidBegin]):
(-[WKWebView _navigationGestureDidEnd]):
(-[WKWebView _setAvoidsUnsafeArea:]):
(-[WKWebView _beginLiveResize]):
(-[WKWebView _endLiveResize]):
(-[WKWebView _safeAreaShouldAffectObscuredInsets]):
(-[WKWebView _beginAnimatedResizeWithUpdates:]):
(-[WKWebView _endAnimatedResize]):
(-[WKWebView _resizeWhileHidingContentWithUpdates:]):
(-[WKWebView _addLayerForFindOverlay]):
(-[WKWebView _removeLayerForFindOverlay]):
(-[WKWebView _layerForFindOverlay]):

Canonical link: <a href="https://commits.webkit.org/253459@main">https://commits.webkit.org/253459@main</a>
</pre>

cdumez reviewed (approved):
LGTM, I think this is a good idea.

smfr committed:
Group WKWebView state which is per-process into a struct
https://bugs.webkit.org/show_bug.cgi?id=243704

Reviewed by Chris Dumez.

We often have bugs where we forget to reset state on WKWebView when the web process terminates (the
same is true on WebPageProxy, which I will fix separately).

To make it harder to create such bugs, group all the data that needs to be reset on web process
termination into a PerWebProcessState struct. `_processWillSwapOrDidExit` can then just reinitialize
this struct.

Currently PerWebProcessState is only used for iOS, but it's probable that future refactoring will
make it apply to all platforms.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
(-[WKWebView _internalDoAfterNextPresentationUpdate:withoutWaitingForPainting:withoutWaitingForAnimatedResize:]):
(-[WKWebView _setMinimumEffectiveDeviceWidth:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _setHasCustomContentView:loadedMIMEType:]):
(-[WKWebView _willInvokeUIScrollViewDelegateCallback]):
(-[WKWebView _didInvokeUIScrollViewDelegateCallback]):
(-[WKWebView _resetCachedScrollViewBackgroundColor]):
(-[WKWebView _updateScrollViewBackground]):
(-[WKWebView _processWillSwapOrDidExit]):
(-[WKWebView _didRelaunchProcess]):
(-[WKWebView _didCommitLoadForMainFrame]):
(-[WKWebView _didCommitLayerTreeDuringAnimatedResize:]):
(-[WKWebView _trackTransactionCommit:]):
(-[WKWebView _restoreScrollAndZoomStateForTransaction:]):
(-[WKWebView _didCommitLayerTree:]):
(-[WKWebView _layerTreeCommitComplete]):
(-[WKWebView _restorePageScrollPosition:scrollOrigin:previousObscuredInset:scale:]):
(-[WKWebView _restorePageStateToUnobscuredCenter:scale:]):
(-[WKWebView _scrollToContentScrollPosition:scrollOrigin:animated:]):
(-[WKWebView _zoomToInitialScaleWithOrigin:animated:]):
(-[WKWebView _initialScaleFactor]):
(-[WKWebView _allowsDoubleTapGestures]):
(-[WKWebView _visibleContentRect]):
(-[WKWebView _dispatchSetViewLayoutSize:]):
(-[WKWebView _dispatchSetDeviceOrientation:]):
(-[WKWebView _beginAutomaticLiveResizeIfNeeded]):
(-[WKWebView _updateLiveResizeTransform]):
(-[WKWebView _frameOrBoundsChanged]):
(-[WKWebView _addUpdateVisibleContentRectPreCommitHandler]):
(-[WKWebView _scheduleVisibleContentRectUpdateAfterScrollInView:]):
(-[WKWebView _shouldDeferGeometryUpdates]):
(-[WKWebView _updateVisibleContentRects]):
(-[WKWebView _cancelAnimatedResize]):
(-[WKWebView _didCompleteAnimatedResize]):
(-[WKWebView _didStopDeferringGeometryUpdates]):
(-[WKWebView _keyboardChangedWithInfo:adjustScrollView:]):
(-[WKWebView _navigationGestureDidBegin]):
(-[WKWebView _navigationGestureDidEnd]):
(-[WKWebView _setAvoidsUnsafeArea:]):
(-[WKWebView _beginLiveResize]):
(-[WKWebView _endLiveResize]):
(-[WKWebView _safeAreaShouldAffectObscuredInsets]):
(-[WKWebView _beginAnimatedResizeWithUpdates:]):
(-[WKWebView _endAnimatedResize]):
(-[WKWebView _resizeWhileHidingContentWithUpdates:]):
(-[WKWebView _addLayerForFindOverlay]):
(-[WKWebView _removeLayerForFindOverlay]):
(-[WKWebView _layerForFindOverlay]):

Canonical link: https://commits.webkit.org/253459@main
